docs/roman/datamodels/structure.rst

```diff
@@ -23,11 +23,11 @@ attribute is not defined, which in turn calls _make_default_array if
 the schema says the attribute is an array. All these methods can be
 found in properties.py.
 
-The base class for Datamodels isDataModel,  in model_base.py. It takes
+The base class for Datamodels is RomanDataModel,  in model_base.py. It takes
 several arguments, the most important of which is init, which as the
 name suggests, specifies how to initialize the primary data array of
 the model. Init is most usually the name of a file, but can be an
-already opened fits or asdf file, a numpy array, a shape tuple, or
+already opened asdf file, a numpy array, a shape tuple, or
```

Comment:
Not for a change here but init is not working unless it is a filename,
>>> init
(200, 200)
>>> t1 = datamodels.open(init)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ddavis/src/Roman/romancal/romancal/datamodels/open_impl.py", line 54, in open
    raise TypeError("init must be a path to an ASDF file or an open AsdfFile instance")
TypeError: init must be a path to an ASDF file or an open AsdfFile instance
or
t1 = datamodels.open(np.array([200,200]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ddavis/src/Roman/romancal/romancal/datamodels/open_impl.py", line 54, in open
    raise TypeError("init must be a path to an ASDF file or an open AsdfFile instance")
TypeError: init must be a path to an ASDF file or an open AsdfFile instance

Reply:
IMO a nice division of responsibility is to have `datamodels.open` handle file paths and `DataModel` handle everything else.  I'm not sure we need `datamodels.open` to be such a swiss army knife, it complicates that function and makes the intention of code that uses it unclear.

I would also like to see `DataModel.__init__` broken down into several class methods:

- `Datamodel.from_shape`
- `DataModel.from_ndarray`
- `DataModel.from_asdf`
- `DataModel.from_fits`
- (etc)

so that the `__init__` function becomes less of a tangle.